### DISCUSSION TITLE
Upgrading to latest Teslamate

### DISCUSSION THREAD

**Discussion by rameshputrevu** [teslamate-org/teslamate#2767] [Q&A]
I have a very old version of Teslamate running on Raspberry 32 bit which is running excellent. Now I am planning to upgrade to Raspberry 64 bit. Would upgrading to the latest version of Telsamate require me entering Access/Refresh token for every 45 days? 

Appreciate your help.

**Comment by cwanja** [comment]
I do not believe the architecture of the Pi effects your tokens. TeslaMate stores a refresh token to continue keeping you logged in.

**Reply by rameshputrevu** [reply]
Thank you, so much.. Just to confirm - refreshing an access token doesn't require captcha/2FA, right? 

So the only hassle is to providing the Access Token and Refresh Token for the first time after we upgrade to latest TeslaMate that uses new V3 auth APIs, is my understanding correct?

**Reply by cwanja** [reply]
Correct, to get refreshed tokens, there is no user interaction at all. You can see that in the logs of the application, it schedules a refresh of the tokens every six hours (if I remember correctly).

**Reply by rameshputrevu** [reply]
Thanks again

**Reply by natrlhy** [reply]
I'm having issues with the `ENCRYPTION_KEY=` line in my docker-compose.yml file. I have a long string there and I still have to login each time with an access and refresh token. Do I need to have it in quotes or anything? it seems like it doesn't like special characters.

**Reply by rameshputrevu** [reply]
After upgrading and restoring my backup file, all my grafana boards are giving this error 
status:500
statusText:"Internal Server Error"
can anyone guide me to troubleshoot this?
